src/psyclone/psyir/tools/dependency_tools.py

```diff
@@ -806,3 +811,208 @@ def can_loop_be_parallelised(self, loop,
                 result = False
 
         return result
+
+# -------------------------------------------------------------------------
+    def can_loops_be_fused(self, loop1, loop2):
+        '''Function that verifies if two loops can be fused.
+
+        :param loop1: the first loop.
+        :type loop1: :py:class:`psyclone.psyir.nodes.Loop`
+        :param loop1: the first loop.
+        :type loop1: :py:class:`psyclone.psyir.nodes.Loop`
+
+        :return: whether the loops can be fused or not.
+        :rtype: bool
+
+        '''
+        # pylint: disable=too-many-locals
+        # This should only be called from loop_fuse_trans, which
+        # has done tests for loop boundaries (depending on domain)
+
+        self._clear_messages()
+        assert isinstance(loop1, Loop)
```

Comment:
Either we assume that the arguments are correct (which I'm prepared to accept) or we should check properly and raise an appropriate error.